### DISCUSSION TITLE
API PNG export option

### DISCUSSION THREAD

**Discussion by Maurice-De** [mattieserver/netbox-topology-views#566] [Ideas]
Hi all,

I am looking for an API export option for PNG files like the XML export function that is already in place. My use case is that I want to include the topologies in offline documentation documents and regularly update those static images with the latest data from Netbox. This helps to prevent that diagrams are made once and never being updated.

So I want to make a small script that gathers all the necessary PNG images and put it in the documentation folder so when someone opens the documentation with a reference to one of the images, they see the latest version.

So if I can make a call to an (API) URL with all the parameters like the XML export API call, but instead of the XML I get an PNG file, that would be nice.

**Comment by mattieserver** [comment]
So we have the "Download PNG" button but that is client side.
I currently don't see a way to create the PNG in the 'backend' so we can return it with an API call.

Depending on your own scripting skills you could use selenium to get the PNG from the "Download PNG" button.

**Reply by Maurice-De** [reply]
Ok, after my vacation I will check if I can make some sort of script for this, but the PNG rendering is done on the client side when you click on the button?

Maybe I need another discussion for this, but is it possible to make a drawing with VLAN ID's shown as a "device" icon in the topology view? For example: I want a topology view with my firewalls and which VLAN is connected to what firewall and the correlations between the VLANs and firewalls.

**Reply by dreng** [reply]
> but the PNG rendering is done on the client side when you click on the button?

Yes. It's a function that belongs to the vis-network library we use for rendering the topology. Vis-network is written in Javascript.

The XML export uses a completely different approach. It calls a URL (similar to the URL of your current view) and creates all needed data in Python. Instead of passing the data to vis-network, it is being used to generate XML code (in Python) and initiates the download.

> For example: I want a topology view with my firewalls and which VLAN is connected to what firewall and the correlations between the VLANs and firewalls.

Can you provide a mockup of how you would expect it to look like? We've had a few discussions and proposals about VLANs before. The problem is that we only display devices and cables. A cable (aka a port) can easily have dozens of VLANs. Because of overlapping labels, that would lead to a very ugly topology very quickly.
